nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/notification/CommentNotificationVeto.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.platform.comment.notification;
+
+import static org.nuxeo.ecm.platform.comment.api.AnnotationConstants.ANNOTATION_DOC_TYPE;
+import static org.nuxeo.ecm.platform.comment.workflow.utils.CommentsConstants.COMMENT_DOC_TYPE;
+
+import java.util.Set;
+
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.ec.notification.NotificationListenerVeto;
+
+/**
+ * Veto that prevents sending notifications whose type is {@link #getExcludedEventType} on the document comment model.
+ * 
+ * @since 11.1
+ */
+public interface CommentNotificationVeto extends NotificationListenerVeto {
+
+    Set<String> ACCEPTED_COMMENT_TYPE = Set.of(COMMENT_DOC_TYPE, ANNOTATION_DOC_TYPE, "Post");
+
+    /**
+     * @return the event type that we don't want to notify
+     */
+    String getExcludedEventType();
+
+    @Override
+    default boolean accept(Event event) {
+        if (!(event.getContext() instanceof DocumentEventContext)) {
+            return false;
+        }
+        DocumentEventContext docCtx = (DocumentEventContext) event.getContext();
+        String eventType = getExcludedEventType();
+        String docType = docCtx.getSourceDocument().getType();
+
+        return !(eventType.equals(event.getName()) && (ACCEPTED_COMMENT_TYPE.contains(docType)));
+    }
+}
```

Comment:
since `default` method mainly i used abstract class when i need some injection if not i used the interface.

Reply:
Ok, got it. Interfaces are more a contract than abstract class. In this though, `ACCEPTED_COMMENT_TYPE` is now part of the API, whereas it looks like an implementation details to me, it's only useful for the current accept implementation. The accept implementation argument also applies to `getExcludedEventType` which is not used outside of this context, and it has the same semantic than `ACCEPTED_COMMENT_TYPE` but is not expressed in the same way, which is odd.

Furthermore, as it is an interface, I'm assuming that I need to use it (implement) if I want to add a notification veto on comments, which will force me to implement `getExcludedEventType` and override the `accept` method to provide my custom logic. In the end, I didn't benefit from the interface + it makes me use it due to its nature.

In case it's an abstract, I know that's an implementation and not a contract to implement. Which drives me to the real interface to implement (and I can still use the abstract if it makes sense).

Also, I'm wondering if we need three distincts class for that? Does only one veto implementation checking the 3 cases is enough?